v2/sourcedb-to-spanner/terraform/samples/sharded-bulk-migration/README.md

```diff
@@ -213,6 +213,37 @@ There can be multiple reasons for this.
     - Consider bumping up `maxConnection` and `numPartition` parameters when launching the jobs. We recommend `320` for
       maxConnections and `10000` for numPartitions.
 
+### Allowing network connectivity
+
+Ensure Dataflow VMs are able to access the MySQL instance. This would require:
+
+- Configuring MySQL to allow connections from the Dataflow IPs.
+- Configuring the firewall to allow ingress TCP connection on the MySQL port
+  from the Dataflow IPs.
+
+#### Configuring connectivity on Cloud SQL
+
+Cloud SQL allows multiple ways to establish connectivity. With google
+private access enabled, Dataflow should automatically be able to [access
+the instance](https://cloud.google.com/sql/docs/mysql/authorize-networks).
+
+#### Configuring connectivity on MySQL on GCE
+
+We recommend running such instances and dataflow inside a VPC ensuring Google
+private access is enabled. By default,
+the firewall is configured to allow connections between all private IPs.
+If the target does not allow the dataflow IPs, we recommend the following configurations
+for allowing connectivity:
+
+- Using network tags to configure MySQL VM. Apply the `databases` network
+  tag on the GCE VM.
+- Add firewall rule that allows tcp ingress on 3306 from resources
+  with the `dataflow` [network tag](https://cloud.google.com/vpc/docs/add-remove-network-tags) on targets
+  with the `databases` tag.
+- Specify the Dataflow ips in source ranges to be allowlisted in the firewall rule.
```

Comment:
What will this be? Dataflow runs a large number of individual GCP VMs and how would the customer know which one will these be?
I think you are confusing how network tags and source ranges work - please check once..

Reply:
Removed the Ip line. Its not needed